### ISSUE TITLE
Find a consistent naming convention

### ISSUE THREAD

**Issue by gracicot** [gracicot/kangaru#46]
A lot of names in the `kgr` namespace is either in camel case, or snake case. Even types and aliases are not consistent in their naming.

While changing naming convention, we propose to use the same naming convention as Boost and STL. So such code:
```c++
struct SomeService : kgr::Service<SomeClass, kgr::Dependency<OtherService>> {};

kgr::Container c;
auto i = c.service<kgr::InvokerService>();
```
will become:
```c++
struct SomeService : kgr::service<SomeClass, kgr::dependency<OtherService>> {};

kgr::container c;
auto i = c.service<kgr::invoker_service>();
```

To reduce transition friction, we propose to write a migration guide, and a shell script to help transition to the new version.

**Comment by gracicot** [comment]
A compatibility header with aliases could be enough.

**Comment by gracicot** [comment]
Fixed in `dev-4.0.x`. A compatibility header is also provided, and all aliases, member alias and types all follow the same naming rules.
